### ISSUE TITLE
[FEATURE] Improve JDBC urls format for Oracle, MSSQL, Teradata

### ISSUE THREAD

**Issue by dolfinus** [OpenLineage/OpenLineage#2762]
### Motivation

Currently OpenLineage integration for Spark supports JDBC urls with following formats:
* `jdbc:type://host:port` - dataset has hame `schema.table`
* `jdbc:type://host:port/database` - database is extracted and used as prefix for dataset name, e.g. `database.schema.table`

Query params (everything after `?`) are trimmed off, as well as user & password parts of URL.
[events_mysql.json](https://github.com/user-attachments/files/15618079/events_mysql.json)

But some databases have different JDBC url formats:
* MSSQL - `jdbc:sqlserver://host:port;database=database` or `jdbc:sqlserver://host:port;databaseName=database` (see [documentation](https://learn.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver16))
* Teradata - `jdbc:teradata://host/DBS_PORT=port,DATABASE=database` (see [documentation](https://teradata-docs.s3.amazonaws.com/doc/connectivity/jdbc/reference/current/jdbcug_chapter_2.html#URL_DATABASE))
* Oracle - `jdbc:oracle:thin:@host:port:sid` or `jdbc:oracle:thin:@//host:port/service` (not mentioning [TNS notation](https://docs.oracle.com/en/database/oracle/oracle-database/21/jjdbc/data-sources-and-URLs.html#GUID-6F729E4D-064B-4FD9-AE92-1BD44B8BE5EF), it is hard to parse).

In this case, OpenLineage integration returns datasets like:
* `namespace=sqlserver://localhost:1433;databaseName=database`, `name=schema.table` [events_mssql.json](https://github.com/user-attachments/files/15618131/events_mssql.json)
* `namespace=teradata://host/DBS_PORT=port,DATABASE=database`, `name=schema.table` (sorry, no example of events for now)
* `namespace=oracle:thin:@host:port:sid`, `name=schema.table` [events_oracle.json](https://github.com/user-attachments/files/15618140/events_oracle.json)

This leads to producing different dataset namespaces for the different JDBC URLs, although some of them are pointing to the same database, just in different connection formats.
More than that, because other connection params are not stripped away, they are also included to dataset namespace. So if user passed custom connection option, like timeout, this will lead to adding one more dataset with a new namespace.

### Description

Proposal - handle those URLs like:
* `namespace=sqlserver://localhost:1433`, `name=database.schema.table` (or `mssql://`)
* `namespace=teradata://host:port`, `name=database.schema.table`
* `namespace=oracle://host:port`, `name=sid.schema.table`

This can be done by:
* Looking up for database name using the same mechanism used for stripping up `user=` and `password=` options from JDBC URL.
* Treat `;` just like `?`. Also if `,` is not a part of netloc, treat it as `?` too.
* Add specific functions for databases like Oracle, which can be expanded in the future (e.g. to add support of TNS format)

### Do you plan to make this contribution yourself?

- [X] I am interested in doing this work

**Comment by dolfinus** [comment]
Regarding MSSQL, user can connect to the instance using port number `jdbc:sqlserver://host:port` or instance name `jdbc:sqlserver://host\instance`. Should the latter be converted to `sqlserver://host\instance` or `sqlserver://host/instance`?

**Comment by dolfinus** [comment]
Preparatory work & implementation for SqlServer: #2773

**Comment by mobuchowski** [comment]
Makes sense.

**Comment by dolfinus** [comment]
Related PRs are merged, thanks!
